dataset/README.md

```diff
@@ -0,0 +1,39 @@
+# User Study Dataset
+This folder contains the various scripts and datasets to produce or replicate the taskset used by the user study. `requirements.txt` contains the python dependencies for the scripts.
+
+# Sources
+The tasks are sourced from StackOverflow, SuperUser, Unix&Linux, CommandLineFu, and Bash One-Liners. 
+For each source, we retrieve the most popular questions or one-liners up to 500 items.
+We retrieve the data from the StackExchange websites (i.e., StackOverflow, SuperUser, Unix&Linux) using the [Stack Exchange Data Explorer](https://data.stackexchange.com/). For CommandLineFu and Bash One-Liners, we use the web scraping scripts developed in `web-scrapers/`.
+
+## SQL Query
+The SQL query used for retrieving the data from the stack exchange website is given below. 
+
+```sql
+SELECT TOP 500 q.Title, q.Id, q.Tags, q.Score, a.Body
+FROM Posts q
+INNER JOIN PostTags pt ON q.Id = pt.PostId
+INNER JOIN Tags t ON pt.TagId = t.Id
+INNER JOIN Posts a ON a.Id = q.AcceptedAnswerId 
+WHERE t.TagName LIKE 'bash' 
+AND q.AcceptedAnswerId is not NULL
+ORDER BY q.Score DESC
+```
+
+The query will return a quintuplet of the question's title, id (used to create a link to the question), tags, score, and raw html body. The `INNER JOIN` on `PostTags` and `Tags` enable to filter questions that are tagged with 'bash' and the last `INNER JOIN` on Posts enable to filter questions that have an accepted answer for the top 500 questions ordered by score.
+
+## Manual exploration
+The web scraping and SQL query are based on available webpages for each site:
+* Stack Overflow: https://stackoverflow.com/search?tab=Votes&q=%5bbash%5d%20hasaccepted%3ayes
+* Super User: https://superuser.com/search?tab=votes&q=%5bbash%5d%20hasaccepted%3ayes
+* Unix & Linux: https://unix.stackexchange.com/questions/tagged/bash?tab=Votes
+* CommandLineFu: https://www.commandlinefu.com/commands/browse/sort-by-votes
+* Bash One-Liners: http://www.bashoneliners.com/oneliners/popular/
+
+# Data
+The collected data from the previous paragraph is stored in the folder `data-collected`. This data is unclean and unfiltered. It totals 2287 records (Bash One-Liners has only 287 records for the entire website). We verified that the most popular questions have at score of at least 0.
+
+## Reconstructing URLs
+For CommandLineFu and Bash One-Liners: 
+* `sed -e 's#^#http://www.bashoneliners.com#' data-collected/bashoneliners-top500.csv > bashoneliners-top500-clean.csv`
+* `sed -e 's#^#https://www.commandlinefu.com#' data-collected/commandlinefu-top500.csv > commandlinefu-top500-clean.csv`
```

Comment:
A small test script to test these `sed` commands would also be helpful. Since they're `Bash`, we can use Bats to test it.

Reply:
Thank you for the suggestion. I wonder what value we would get out of it?

Reply:
Seds and regexes are just generally icky, I suppose reading a test case with what the input/expected output would be helps explains these kinds of stuff. If it's a bit too complicated, maybe just a quick comment on what you're stripping would be nice. 

Now that I'm looking at the `sed` commands mentioned again, I'm actually not quite sure what they're doing. It looks like they're just matching the URLs without doing any replacement, in which case, why can't we use `grep -e` here?